src/VisualStudio/Core/Def/Implementation/ColorSchemes/ColorSchemeApplier.ClassificationVerifier.cs

```diff
@@ -0,0 +1,212 @@
+                    [ClassificationTypeNames.Identifier] = LightThemeIdentifier,
+                    [ClassificationTypeNames.Keyword] = LightThemeKeyword,
+                    [ClassificationTypeNames.Operator] = LightThemeOperator,
+                }.ToImmutableDictionary();
+
+            // The High Contrast theme is not included because we do not want to make changes when the user is in High Contrast mode.
+
+            private IVsFontAndColorStorage? _fontAndColorStorage;
+            private IVsFontAndColorUtilities? _fontAndColorUtilities;
+
+            private ImmutableArray<string> Classifications { get; }
+
+            public ClassificationVerifier(IThreadingContext threadingContext, IServiceProvider serviceProvider, ImmutableDictionary<SchemeName, ColorScheme> colorSchemes)
+                : base(threadingContext)
+            {
+                _serviceProvider = serviceProvider;
+
+                _colorSchemes = colorSchemes.ToImmutableDictionary(
+                    nameAndScheme => nameAndScheme.Key,
+                    nameAndScheme => nameAndScheme.Value.Themes.ToImmutableDictionary(
+                        theme => theme.Guid,
+                        theme => theme.Category.Colors
+                            .Where(color => color.Foreground.HasValue)
+                            .ToImmutableDictionary(color => color.Name, color => color.Foreground!.Value)));
+
+                // Gather all the classifications from the core and scheme dictionaries.
+                var coreClassifications = DarkThemeForeground.Keys.Concat(BlueLightThemeForeground.Keys).Distinct();
+                var colorSchemeClassifications = _colorSchemes.Values.SelectMany(scheme => scheme.Values.SelectMany(theme => theme.Keys)).Distinct();
+                Classifications = coreClassifications.Concat(colorSchemeClassifications).ToImmutableArray();
+            }
+
+            /// <summary>
+            /// Determines if any Classification foreground colors have been customized in Fonts and Colors.
+            /// </summary>
+            public bool AreClassificationsCustomized(SchemeName schemeName, Guid themeId)
+            {
+                AssertIsForeground();
+
+                // Ensure we are initialized
+                if (_fontAndColorStorage is null)
+                {
+                    _fontAndColorStorage = _serviceProvider.GetService<SVsFontAndColorStorage, IVsFontAndColorStorage>();
+                    _fontAndColorUtilities = (IVsFontAndColorUtilities)_fontAndColorStorage!;
+                }
+
+                // Make no changes when in high contast mode or in unknown theme.
+                if (SystemParameters.HighContrast ||
+                    !_colorSchemes.TryGetValue(schemeName, out var colorScheme) ||
+                    !colorScheme.ContainsKey(themeId))
+                {
+                    return false;
+                }
+
+                var coreThemeColors = (themeId == KnownColorThemes.Dark)
```

Comment:
❓ Are `Blue` and `BlueLight` the same here?

Reply:
Yup, Blue and Light share the same editor colors.